src/open_producten/utils/drf_validators.py

```diff
@@ -19,3 +24,60 @@ def __call__(self, value, serializer):
 
         if errors:
             raise serializers.ValidationError(errors)
+
+
+class NestedObjectsValidator:
+    requires_context = True
+
+    def __init__(self, key: str, model: Type[models.Model]):
+        self.key = key
+        self.model = model
+
+    def __call__(self, value, serializer):
+        parent_instance = serializer.instance
+
+        if not parent_instance or not value.get(self.key):
+            return
+
+        errors = []
+
+        current_ids = set(
+            getattr(parent_instance, self.key).values_list("id", flat=True).distinct()
+        )
+
+        seen_ids = set()
+
+        for idx, obj in enumerate(value[self.key]):
+            obj_id = obj.get("id", None)
+
+            if not obj_id:
+                continue
+
+            if obj_id in current_ids:
+
+                if obj_id in seen_ids:
+                    errors.append(_("Dubbel id: {} op index {}.").format(obj_id, idx))
+                seen_ids.add(obj_id)
+
+            else:
+                try:
+                    self.model.objects.get(id=obj_id)
```

Comment:
Can you add a comment to explain that it is not allowed to modify objects that are not related to the parent object? I was a bit confused initially as to why an error is raised inside the try block